client/src/featureform/serving.py

```diff
@@ -1002,10 +1005,23 @@ def dataframe(self) -> pd.DataFrame:
         ```
 
         Returns:
-            pandas.DataFrame: A pandas DataFrame containing the training set.
+            df: Union[pd.DataFrame, pyspark.sql.DataFrame] A DataFrame containing the training set.
         """
+
         if self._dataframe is not None:
             return self._dataframe
+        elif spark_session is not None:
+            req = serving_pb2.TrainingDataRequest()
+            req.id.name = self._stream.name
+            req.id.version = self._stream.version
+            resp = self._stream._stub.ResourceLocation(req)
```

Comment:
nit: this should probably be it's own method somewhere so that anyone can use this API